src/utilities.ts

```diff
@@ -43,6 +44,50 @@ export function getMinorFullName(acronym: string): string {
   return requirementJSON.minor[acronym].name;
 }
 
+export function getAllSubjects(): ReadonlySet<string> {
+  const set = new Set<string>();
+  fullCoursesArray.forEach(it => set.add(it.subject));
+  return set;
+}
+
+const SUBJECT_COLORS = [
+  {
+    text: 'Red',
+    hex: 'DA4A4A',
+  },
+  {
+    text: 'Orange',
+    hex: 'FFA53C',
+  },
+  {
+    text: 'Green',
+    hex: '58C913',
+  },
+  {
+    text: 'Blue',
+    hex: '139DC9',
+  },
+  {
+    text: 'Purple',
+    hex: 'C478FF',
+  },
+  {
+    text: 'Pink',
+    hex: 'F296D3',
+  },
+];
+
+export function allocateSubjectColor(subjectColors: Record<string, string>, subject: string): void {
+  if (subjectColors[subject]) return;
+  const usedColors = new Set(Object.values(subjectColors));
+  // Filter out used colors
+  const unusedSubjectColors = SUBJECT_COLORS.filter(color => !usedColors.has(color.hex));
+  const subjectColorCandidates =
+    unusedSubjectColors.length !== 0 ? unusedSubjectColors : SUBJECT_COLORS;
+  subjectColors[subject] =
+    subjectColorCandidates[Math.floor(Math.random() * subjectColorCandidates.length)].hex;
```

Comment:
What is the difference between this and assigning a random color to every subject?
It seems like `subjectColorCandidates` is only `unusedSubjectColors` for the first 6 subjects, and then colors are assigned randomly from then on.

Reply:
There might be no difference on the larger scale. Maybe we can just randomize everything since the first 6 subjects means nothing now...

Reply:
Yeah could we possibly just randomize everything? I agree that there prob wouldn't be much meaning to the first 6 subjects that have colors assigned, especially if it ends up just being the first 6 subjects alphabetically. My only note in that case would be that the test would prob have to change. The best we could prob do for the test would be to make sure that all assigned colors are elements of `SUBJECT_COLORS` unless there is a smarter test we can think of..